### PR TITLE
Replace travis with github actions build

### PR DESCRIPTION
**Pull request by SimonIT**
This is basically the same build which currently runs on travis.

We have to wait probably on #5976 or #6240, because the current build fails without the fixes included there.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+      - name: Setup ccache
+        run: |
+          brew install ccache
+          echo "/usr/local/opt/ccache/libexec" >> $GITHUB_PATH
+      - name: Build macOS natives
+        run: ant -f build-mac-ios.xml
+
+  linux:
+    needs: macos
```
Comment (PokeMMO) [inline .github/workflows/build.yml:36]: Since this does not depend on uploading macOS natives for linux to use, we do not need this line.

**Review by PokeMMO — APPROVED** [review]

**Comment by PokeMMO** [comment]
https://github.com/libgdx/libgdx/actions/runs/320268639

There's some deprecation warnings that we should fix.

**Comment by SimonIT** [comment]
I saw them. The warnings are from the setup-android action. I don't know how to fix them because the functions are used internally

**Comment by PokeMMO** [comment]
Oh, sorry, you're right. Nothing for us to do there then.

**Pull request merged**